sql/core/src/main/scala/org/apache/spark/sql/execution/WholeStageCodegenExec.scala

```diff
@@ -527,12 +526,29 @@ case class WholeStageCodegenRDD(@transient sc: SparkContext, var source: CodeAnd
       } catch {
         case e: Throwable =>
           if (WholeStageCodegenExec.dumpGenCodeForException) {
-            logError(s"\n${CodeFormatter.format(source)}")
+            logFormattedError(e, s"\n${CodeFormatter.format(source)}")
           }
           throw e
       }
+    }
+  }
 
+  def logFormattedError(e: Throwable, source: String): Unit = {
+    var cause = e
+    while (cause ne null) {
+      // Don't log the code when the exception is out of memory
+      cause match {
+        case e: SQLException if e.getSQLState == "XCL54.T" =>
+          return
+        case o: RuntimeException if o.getClass.equals(
+          // scalastyle:off classforname
+          Class.forName("com.gemstone.gemfire.cache.LowMemoryException")) =>
```

Comment:
Check name of class rather than Class.forName that will fail with ClassNotFoundException in pure Spark mode: `e.getClass.getName.contains("LowMemoryException")`